SOURCES/etc/xapi.d/plugins/updater.py

```diff
@@ -75,6 +78,54 @@ def upgrade():
     pass
 
 
+CONFIGURATION_FILE = '/etc/yum.repos.d/xcp-ng.repo'
+
+
+# returns a JSON dict {repo_id: proxy}
+def get_proxies(session, args):
+    config = ConfigParser.ConfigParser({'proxy': '_none_'})
+    config.read(CONFIGURATION_FILE)
+    new_dict = dict((section, config.get(section, 'proxy', '_none_')) for section in config.sections())
+    return json.dumps(new_dict)
```

Comment:
This function could fail for various reasons: missing file, malformed configuration file, I/O errors. Wouldn't it be safer to return a status too? Also, maybe we should kind of standardize the return types (a dict that always contains a 'status' key and optional 'error': key for error text) in this plugin and return with a `false` status if anything goes wrong? Or is there already some sort of automated error handling that informs the caller of an error in case of unhandled exceptions?

Reply:
actually it just returns `{}` if it can't read the file, `RawConfigParser.read()` doesn't throw errors, it just returns the list of successfully read files.

Reply:
Then we should document that behaviour.